tests/test_peaks.py

```diff
@@ -0,0 +1,71 @@
+import strax
+import straxen
+
+import numpy as np
+
+import unittest
+from strax.testutils import run_id
+from hypothesis import strategies, given, settings
+
+TEST_DATA_LENGTH = 3
+
+
+class TestComputePeakBasics(unittest.TestCase):
+    """Tests for peak basics plugin"""
+
+    def setUp(self, context=straxen.contexts.demo):
+        self.st = context()
+        self.n_top = self.st.config.get('n_top_pmts', 2)
+
+        # Make sure that the check is on. Otherwise we cannot test it.
+        self.st.set_config({'check_peak_sum_area': True})
+        self.peaks_basics_compute = self.st.get_single_plugin(run_id, 'peak_basics').compute
+
+    @settings(deadline=None)
+    @given(strategies.integers(min_value=0,
+                               max_value=TEST_DATA_LENGTH - 1),
+           )
+    def test_aft_equals1(self, test_peak_idx):
+        """Fill top array with area 1"""
+        test_data = self.get_test_peaks(self.n_top)
+        test_data[test_peak_idx]['area_per_channel'][:self.n_top] = 1
+        test_data[test_peak_idx]['area'] = np.sum(test_data[test_peak_idx]['area_per_channel'])
+        peaks = self.peaks_basics_compute(test_data)
+        assert peaks[test_peak_idx]['area_fraction_top'] == 1
+
+    @settings(deadline=None)
+    @given(strategies.floats(min_value=0,
+                             max_value=2,
+                             ),
+           strategies.integers(min_value=0,
+                               max_value=TEST_DATA_LENGTH - 1),
+           )

```

Comment:
You could also filter the strategies itself. In that way you make sure that you do not "test" a case which is not actually tested in the end. I am refering to 
```
if np.isclose(off_by_factor, 1) or np.isclose(off_by_factor, 0):
            return```

Reply:
Sure, the advantage being that we are not supplying meaningless numbers to the test.

I've added this in .filter(_not_close_to_0_or_1)